include/zephyr/bluetooth/classic/classic.h

```diff
@@ -0,0 +1,197 @@
+/** @file
+ *  @brief Bluetooth subsystem classic core APIs.
+ */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_CLASSIC_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_CLASSIC_H_
+
+/**
+ * @brief Bluetooth APIs
+ * @defgroup bluetooth Bluetooth APIs
+ * @ingroup connectivity
+ * @{
+ */
+
+#include <stdbool.h>
+#include <string.h>
+
+#include <zephyr/sys/util.h>
+#include <zephyr/net_buf.h>
+#include <zephyr/bluetooth/addr.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Generic Access Profile (GAP)
+ * @defgroup bt_gap Generic Access Profile (GAP)
+ * @since 1.0
+ * @version 1.0.0
+ * @ingroup bluetooth
+ * @{
+ */
+
+/** @brief BR/EDR discovery result structure */
+struct bt_br_discovery_result {
+	/** private */
+	uint8_t _priv[4];
```

Comment:
I was just looking at how this is used, and it seems quite fragile. I'd recommend to clean it up in the same go. These four bytes are cast into the internally defined `struct discovery_priv`, but there's not even any build assert to ensure that the sizes match up. I'd suggest to just put the private struct in the public header, but cleanly document that it's private data, i.e. something like:
```c
struct bt_br_discovery_result {
	struct bt_br_discovery_priv _priv;
	...
};
```

Reply:
Updated.